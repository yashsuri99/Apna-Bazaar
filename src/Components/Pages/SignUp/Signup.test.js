import Signin from "./SignUp";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";

describe("Login credentials form on submit", () => {
    const signUp = {
        firstName: "clark",
        lastName: "doe",
        email: "group@sapient.com",
        password: "Password123!",
        confirmPassword: "Password123!",
    };

    let firstNameInput,
        lastNameInput,
        emailInput,
        passwordInput,
        confirmPasswordInput,
        submitButton;

    beforeEach(() => {
        // eslint-disable-next-line testing-library/no-render-in-setup
        render(
            <BrowserRouter>
                <Signin />
            </BrowserRouter>
        );

        firstNameInput = screen.getByLabelText(/First Name/i);
        lastNameInput = screen.getByLabelText(/Last Name/i);
        emailInput = screen.getByLabelText(/Email/i);
        // passwordInput = screen.getAllByLabelText(/Password/i);
        // confirmPasswordInput = screen.getAllByLabelText(/Confirm Password/i);
        submitButton = screen.getByRole("button", { name: /SignUp/i });

        userEvent.clear(firstNameInput);
        userEvent.clear(lastNameInput);
        userEvent.clear(emailInput);
        // userEvent.clear(passwordInput);
        // userEvent.clear(confirmPasswordInput);
    });
    const fillAndSubmit = (credentials) => {
        userEvent.type(firstNameInput, credentials.firstName);
        userEvent.type(lastNameInput, credentials.lastName);
        userEvent.type(emailInput, credentials.email);
        // userEvent.type(passwordInput, credentials.password);
        // userEvent.type(confirmPasswordInput, credentials.confirmPassword);
        userEvent.click(submitButton);
    };

    test("should display error when first name is not filled", async () => {
        fillAndSubmit({
            ...signUp,
            firstName: "",
        });
        const errorMessageEl = await screen.findByText(/First Name required/);
        expect(errorMessageEl).toBeInTheDocument();
        expect(errorMessageEl).toMatchSnapshot();
    });

    test("should display error when last name is not filled", async () => {
        fillAndSubmit({
            ...signUp,
            lastName: "",
        });
        const errorMessageEl = await screen.findByText(/Last Name required/);
        expect(errorMessageEl).toBeInTheDocument();
        expect(errorMessageEl).toMatchSnapshot();
    });

    test("should display error when email is not filled", async () => {
        fillAndSubmit({
            ...signUp,
            email: "",
        });
        const errorMessageEl = await screen.findByText(/Email required/i);
        expect(errorMessageEl).toBeInTheDocument();
        expect(errorMessageEl).toMatchSnapshot();
    });

    // test("should display error when password is not filled", async () => {
    //     fillAndSubmit({
    //         ...signUp,
    //         password: "",
    //     });
    //     const errorMessageEl = await screen.findAllByText(/Password required/i);
    //     expect(errorMessageEl).toBeInTheDocument();
    // });

    // test("should display error when confirm password is not filled", async () => {
    //     fillAndSubmit({
    //         ...signUp,
    //         confirmPassword: "",
    //     });
    //     const errorMessageEl = await screen.findByText(/Confirm Password/i);
    //     expect(errorMessageEl).toBeInTheDocument();
    // });
});
