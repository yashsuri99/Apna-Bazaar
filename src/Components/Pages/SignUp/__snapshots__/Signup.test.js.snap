// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login credentials form on submit should display error when email is not filled 1`] = `
<p
  class="form-error"
>
  Email required
</p>
`;

exports[`Login credentials form on submit should display error when first name is not filled 1`] = `
<p
  class="form-error"
>
  First Name required
</p>
`;

exports[`Login credentials form on submit should display error when last name is not filled 1`] = `
<p
  class="form-error"
>
  Last Name required
</p>
`;
